### PR TITLE
refactor: Remodel NetworkInfoProvider to remove NetworkInfo

### PR DESCRIPTION
**Pull request by IvayloAndonov**
# Context

This actually highlights an opportunity to improve our `NetworkInfoProvider` interface, by exposing `maxLovelaceSupply` and `networkId` under `genesisParameters`. That would leave `NetworkInfo.network.timeSettings`, which we can consider lifting as a new top-level provider method, and then remove the `NetworkInfo` type by hoisting the remaining properties too

# Proposed Solution
See the changes introduced section

# Important Changes Introduced
- (blockfrost): align [blockfrostNetworkInfoProvider](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-72ea4c5bbe10bbaa5b65a22a09b6787002610454ad1ee398543e2f4fc19ee00f) with the improved `NetworkInfoProvider` interface

- (cardano-services-client): extend [networkInfoHttpProvider.ts](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-17aff5486bd220076e5e5777bf82e639421e37f4d204a1a5bc787f636b2057b4) with top-level methods

- (cardano-services): remodel [DbSynNetworkInfoProvider](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-93293b161f83ab2bf95474e66fcb0cd45438d4a6b4775fd2761964eb393902a5), lift methods from `networkInfo`, 

- (cardano-services): extend [NetworkInfoHttpService](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-9d24177fdd0e4472fe97c21580075de9a807cd0866edcb82f888698740c73618) with `/stake`, `lovelace-supply` , `/time-settings`

- (cardano-services): align [openApi](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-7e101204ca1117f9fd739bbf060ecd83683fe997700f6788b51ce3697e510ff4) spec and tests

- (core): add `networkId` as part of [CompactGenesis](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-fa2c462919a323d51156377031b4fc5b363c71feb787bf65d74d3e84cb003c95) interface (`maxLovelaceSupply` is already exposed by it)

- (core): lift `timeSettings`, `stake`, `lovelaceSupply` as [new top-level methods](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-4cbf8347b151acd22f2d393b3fafae8d4e42a2acf138a5048cf73fe0b5abb9d4) and remove `NetworkInfo` from the provider interface, rename `AssetSupply` to `SupplySummary` consistent to `StakeSummary`

- (wallet): refactor all `NetworkInfo` dependent files and tests

- (web-extension): update [util](https://github.com/input-output-hk/cardano-js-sdk/pull/301/files#diff-b5e6537a260f1fdc53e71cea1c4e6064ec89e2ee9682a9d525b099ebcb56ae52) file which depends on `networkInfo`

# Note
This refactoring brings another good advantage for SDK clients - exposing `NetworkInfoProvider` interface methods in a more granular way which provides more flexibility and performance of Network info-related data fetching (fewer DB computations per method)

### CODE REVIEW

File: packages/core/src/Provider/NetworkInfoProvider/types.ts [hunk 1]
```diff
@@ -16,9 +16,8 @@ export type ProtocolParametersRequiredByWallet = Required<
   >
 >;
 
-export type AssetSupply = {
+export type SupplySummary = {
   circulating: Cardano.Lovelace;
-  max: Cardano.Lovelace;
```
Comment (IvayloAndonov) [inline packages/core/src/Provider/NetworkInfoProvider/types.ts:21]: Removed from the interface since it's already exposed by [CompactGenesis](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Cardano/types/Genesis.ts#L9-L19)

File: packages/wallet/src/cip30.ts [hunk 2]
```diff
@@ -105,9 +99,8 @@ export const createWalletApi = (
   getNetworkId: async (): Promise<Cardano.NetworkId> => {
     logger.debug('getting networkId');
     const wallet = await walletReady;
-    const networkInfo = await firstValueFrom(wallet.networkInfo$);
-    // TODO: use networkInfo.network.id instead of magicToNetworkId
-    return magicToNetworkId(networkInfo.network.magic);
```
Comment (IvayloAndonov) [inline packages/wallet/src/cip30.ts:110]: Following the TODO comment, `genesisParameters.networkId` is used directly, removed obsolete `magicToNetworkId` mapping

File: packages/wallet/src/SingleAddressWallet.ts [hunk 3]
```diff
@@ -199,16 +203,27 @@ export class SingleAddressWallet implements ObservableWallet {
...
       // This is a little complicated since there is a circular dependency.
       // Some logic is needed to initiate a fetch if epoch is not available in store already.
-      coldObservableProvider(this.networkInfoProvider.networkInfo, retryBackoffConfig, tipBlockHeight$, deepEquals),
-      stores.networkInfo
+      coldObservableProvider(this.networkInfoProvider.timeSettings, retryBackoffConfig, tipBlockHeight$, deepEquals),
+      stores.timeSettings
     );
-    this.currentEpoch$ = currentEpochTracker(this.tip$, this.networkInfo$);
+    this.currentEpoch$ = currentEpochTracker(this.tip$, this.timeSettings$);
     const epoch$ = this.currentEpoch$.pipe(map((epoch) => epoch.epochNo));
```
[the remark this answers is not in the document]
Comment (IvayloAndonov) [reply packages/wallet/src/SingleAddressWallet.ts:215]: Thanks for the clarification 👍🏻

**Review by mkazlauskas — APPROVED** [review]
Nice work @IvayloAndonov !

Side note: we could probably cache more data (like protocol params), but let's do the optimizations separately if we decide we want to do it at all

**Review by rhyslbw — APPROVED** [review]
:tada:

**Review by rhyslbw — CHANGES_REQUESTED** [review]
I've just noticed the first commit, which is the primary one, is broken. This indicates to me there's a fix in the second that needs to be extracted as a fixup for the first:
image

We'll also need SDET review given the test suite has changed more than just the minimum to align with the source refactor

**Review by rhyslbw — APPROVED** [review]
Nice work @IvayloAndonov. You can now squash ready for SDET review

**Review by ssong-iohk — CHANGES_REQUESTED** [review]

File: packages/wallet/test/persistence/inMemoryStores.test.ts [hunk 4]
```diff
@@ -6,11 +6,13 @@ import {
   InMemoryGenesisParametersStore,
   InMemoryInFlightTransactionsStore,
   InMemoryKeyValueStore,
-  InMemoryNetworkInfoStore,
   InMemoryProtocolParametersStore,
   InMemoryRewardsBalancesStore,
   InMemoryRewardsHistoryStore,
   InMemoryStakePoolsStore,
+  InMemoryStakeSymmaryStore,
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply packages/wallet/test/persistence/inMemoryStores.test.ts:13]: @ssong-iohk try using the suggestion feature of GitHub
image
image

[the remark this answers is not in the document]
Comment (IvayloAndonov) [reply packages/wallet/test/persistence/inMemoryStores.test.ts]: [d63d3fb](https://github.com/input-output-hk/cardano-js-sdk/pull/301/commits/d63d3fbccb414cc75dad8d4a5a43cc611798c281)

**Review by ssong-iohk — APPROVED** [review]
LGTM. thanks a lot!

**Review by rhyslbw — APPROVED** [review]
